projects/igniteui-angular/src/lib/combo/combo.common.ts

```diff
@@ -1211,8 +1211,14 @@ export abstract class IgxComboBaseDirective extends DisplayDensityBase implement
         if (this.comboAPI.valueKey === null) {
             return keys;
         }
+
+        const isNaNvalue = (key: any): boolean => (isNaN(key) && key !== undefined && typeof key !== 'string') ?? false;
```

Comment:
no need to fallback to `false` when `(isNaN(key) && key !== undefined && typeof key !== 'string')` returns `false`.